kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/PodOperationsImpl.java

```diff
@@ -18,60 +18,168 @@
+
+    private static final String[] EMPTY_COMMAND = {"/bin/sh", "-i"};
+
+    private final String containerId;
+    private final InputStream in;
+    private final OutputStream out;
+    private final OutputStream err;
+    private final boolean withTTY;
+
+    public PodOperationsImpl(OkHttpClient client, Config config, String namespace) {
+        this(client, config, null, namespace, null, true, null);
+    }
+
+    public PodOperationsImpl(OkHttpClient client, Config config, String apiVersion, String namespace, String name, Boolean cascading, Pod item) {
+        this(client, config, apiVersion, namespace, name, cascading, item, null, null, null, null, false);
+    }
+
+    public PodOperationsImpl(OkHttpClient client, Config config, String apiVersion, String namespace, String name, Boolean cascading, Pod item, String containerId, InputStream in, OutputStream out, OutputStream err, boolean withTTY) {
+        super(client, config, null, apiVersion, "pods", namespace, name, cascading, item);
+        this.containerId = containerId;
+        this.in = in;
+        this.out = out;
+        this.err = err;
+        this.withTTY = withTTY;
+    }
+
+    @Override
+    public String getLog() {
+        return getLog(true);
+    }
+
+    @Override
+    public String getLog(Boolean isPretty) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("log?pretty=").append(isPretty);
+        if (containerId != null && !containerId.isEmpty()) {
+            sb.append("&container=").append(containerId);
+        }
+
+        try {
+            URL url = new URL(URLUtils.join(getResourceUrl().toString(), sb.toString()));
+            Request.Builder requestBuilder = new Request.Builder().get().url(url);
+            Request request = requestBuilder.build();
+            Response response = client.newCall(request).execute();
+            assertResponseCode(request, response, 200);
+            return response.body().string();
+        } catch (Throwable t) {
+            throw KubernetesClientException.launderThrowable(t);
+        }
+    }
+
+    @Override
+    public String getLog(String containerId) {
+        return new PodOperationsImpl(client, getConfig(), apiVersion, namespace, name, isCascading(), getItem(), containerId, in, out, err, withTTY).getLog(true);
+    }
+
+    @Override
+    public String getLog(String containerId, Boolean isPretty) {
+        return new PodOperationsImpl(client, getConfig(), apiVersion, namespace, name, isCascading(), getItem(), containerId, in, out, err, withTTY).getLog(isPretty);
+    }
+
+    @Override
+    public ContainerResource<String, InputStream, OutputStream, OutputStream, String, Watch> inContainer(String containerId) {
+        return new PodOperationsImpl(client, getConfig(), apiVersion, namespace, name, isCascading(), getItem(), containerId, in, out, err, withTTY);
+    }
+
+    @Override
+    public TtyExecable<String, Watch> usingError(OutputStream err) {
```

Comment:
From my POV its not the same thing as a Process, since we advise exec where to place the output, rather than asking exec from where to read the output.

Still, if it makes life easier to provide "additional" methods for the piping, then why not?